plugins/outputs/newrelic/newrelic.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+)
+
+// NewRelic nr structure
+type NewRelic struct {
+	harvestor    *telemetry.Harvester
+	dc           *cumulative.DeltaCalculator
+	InsightsKey  string            `toml:"insights_key"`
+	MetricPrefix string            `toml:"metric_prefix"`
+	Timeout      internal.Duration `toml:"timeout"`
+	savedErrors  map[int]interface{}
+	errorCount   int
+	Client       http.Client
+}
+
+// Description returns a one-sentence description on the Output
+func (nr *NewRelic) Description() string {
+	return "Send metrics to New Relic metrics endpoint"
+}
+
+// SampleConfig : return  default configuration of the Output
+func (nr *NewRelic) SampleConfig() string {
+	return `
+	## New Relic Insights API key (required)
+	insights_key = "insights api key"
+
+	# metric_prefix if defined, prefix's metrics name for easy identification (optional)
+	# metric_prefix = ""
+	
+	# harvest timeout, default is 15 seconds
+	# timeout = "15s"
+`
+}
+
+// Connect to the Output
+func (nr *NewRelic) Connect() error {
+	if nr.InsightsKey == "" {
+		return fmt.Errorf("InsightKey is a required for newrelic")
+	}
+	var err error
+	nr.harvestor, err = telemetry.NewHarvester(telemetry.ConfigAPIKey(nr.InsightsKey),
+		telemetry.ConfigHarvestPeriod(0),
+		func(cfg *telemetry.Config) {
+			cfg.Product = "NewRelic-Telegraf-Plugin"
+			cfg.ProductVersion = "1.0"
+			cfg.HarvestTimeout = nr.Timeout.Duration
+			cfg.Client = &nr.Client
+			cfg.ErrorLogger = func(e map[string]interface{}) {
+				var errorString string
+				for k, v := range e {
+					errorString += fmt.Sprintf("%s = %s ", k, v)
+				}
+				nr.errorCount++
+				nr.savedErrors[nr.errorCount] = errorString
+			}
+		})
+	if err != nil {
+		return fmt.Errorf("unable to connect to newrelic %v", err)
+	}
+
+	nr.dc = cumulative.NewDeltaCalculator()
+	return nil
+}
+
+// Close any connections to the Output
+func (nr *NewRelic) Close() error {
+
+	nr.errorCount = 0
+	nr.Client.CloseIdleConnections()
+	return nil
+}
+
+// Write takes in group of points to be written to the Output
+func (nr *NewRelic) Write(metrics []telegraf.Metric) error {
+
+	nr.errorCount = 0
+	nr.savedErrors = make(map[int]interface{})
+
+	for _, metric := range metrics {
+		// create tag map
+		tags := make(map[string]interface{})
+		for _, tag := range metric.TagList() {
+			tags[tag.Key] = tag.Value
+		}
+		for _, field := range metric.FieldList() {
+			var mvalue float64
+			var mname string
+			if nr.MetricPrefix != "" {
+				mname = nr.MetricPrefix + "." + metric.Name() + "." + field.Key
+			} else {
+				mname = metric.Name() + "." + field.Key
+			}
+			switch n := field.Value.(type) {
+			case int64:
+				mvalue = float64(n)
+			case uint64:
+				mvalue = float64(n)
+			case float64:
+				mvalue = float64(n)
+			case bool:
+				mvalue = float64(0)
+				if n {
+					mvalue = float64(1)
+				}
+			case string:
+				// Do not log everytime we encounter string
+				// we just skip
```

Comment:
@hsinghkalsi Could you append it to tags here if it's a string?

Reply:
Consider this for a future refactor after we get feedback from the initial users.

Reply:
👍